font-types/src/name_id.rs

```diff
@@ -0,0 +1,184 @@
+/// about the font and those are provided as associated constants on this type.
+///
+/// IDs 26 to 255, inclusive, are reserved for future standard names. IDs 256 to 32767,
+/// inclusive, are reserved for font-specific names such as those referenced by a font’s
+/// layout features.
+///
+/// For more detail, see <https://learn.microsoft.com/en-us/typography/opentype/spec/name#name-ids>
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct NameId(u16);
+
+impl NameId {
+    /// Copyright notice.
+    pub const COPYRIGHT_NOTICE: Self = Self(0);
+
+    /// Font Family name. The Font Family name is used in combination with Font Subfamily name (name ID 2),
+    /// and should be shared among at most four fonts that differ only in weight or style (as described below).
+    pub const FAMILY_NAME: Self = Self(1);
+
+    /// Font Subfamily name. The Font Subfamily name is used in combination with Font Family name (name ID 1),
+    /// and distinguishes the fonts in a group with the same Font Family name. This should be used for style
+    /// and weight variants only (as described below).
+    pub const SUBFAMILY_NAME: Self = Self(2);
+
+    /// Unique font identifier.
+    pub const UNIQUE_ID: Self = Self(3);
+
+    /// Full font name that reflects all family and relevant subfamily descriptors. The full font name is generally
+    /// a combination of name IDs 1 and 2, or of name IDs 16 and 17, or a similar human-readable variant.
+    pub const FULL_NAME: Self = Self(4);
+
+    /// Version string. Should begin with the syntax “Version number.number” (upper case, lower case, or mixed,
+    /// with a space between “Version” and the number).
+    pub const VERSION_STRING: Self = Self(5);
+
+    /// PostScript name for the font; Name ID 6 specifies a string which is used to invoke a PostScript language font
+    /// that corresponds to this OpenType font. When translated to ASCII, the name string must be no longer than 63
+    /// characters and restricted to the printable ASCII subset, codes 33 to 126, except for the 10 characters '[', ']',
+    /// '(', ')', '{', '}', '<', '>', '/', '%'.
+    pub const POSTSCRIPT_NAME: Self = Self(6);
+
+    /// Trademark; this is used to save any trademark notice/information for this font. Such information should be based
+    /// on legal advice. This is distinctly separate from the copyright.
+    pub const TRADEMARK: Self = Self(7);
+
+    /// Manufacturer name.
+    pub const MANUFACTURER: Self = Self(8);
+
+    /// Designer; name of the designer of the typeface.
+    pub const DESIGNER: Self = Self(9);
+
+    /// Description; description of the typeface. Can contain revision information, usage recommendations, history, features, etc.
+    pub const DESCRIPTION: Self = Self(10);
+
+    /// URL Vendor; URL of font vendor (with protocol, e.g., http://, ftp://). If a unique serial number is embedded in the URL,
+    /// it can be used to register the font.
```

Comment:
```suggestion
    /// URL Vendor; URL of font vendor (with protocol, e.g., http://, ftp://). 
    ///
    /// If a unique serial number is embedded in the URL, it can be used to register the font.
```